### PR TITLE
yum/dnf: fail when space separated string of names

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY
@maxamillion I wonder if the failure is too much but the docs clarification is worth adding I think.

Related to https://github.com/ansible/ansible/issues/46301

##### ISSUE TYPE

- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME

yum
dnf

##### ANSIBLE VERSION

```paste below
devel
```

### CODE REVIEW

File: lib/ansible/module_utils/yumdnf.py [hunk 1]
```diff
@@ -98,6 +98,15 @@ def __init__(self, module):
         self.enablerepo = self.listify_comma_sep_strings_in_list(self.enablerepo)
         self.exclude = self.listify_comma_sep_strings_in_list(self.exclude)
 
+        # Fail if someone passed a space separated string
+        # https://github.com/ansible/ansible/issues/46301
+        if any((' ' in name for name in self.names)):
```
Comment (mkrizek) [inline lib/ansible/module_utils/yumdnf.py]: Hrm, so my assumption "who would put space in a package name?" wasn't true of course. This breaks `@Development Tools` use case.

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/88652/35/tests

File: lib/ansible/module_utils/yumdnf.py (hunk 1 above)
Comment (maxamillion) [reply lib/ansible/module_utils/yumdnf.py]: Right, so a rpm package name can *not* contain a space, but groups can. I guess you can just add a `and '@' not in name` to your conditional and that should handle it.

**Comment by maxamillion** [comment]
rebuild_merge

**Comment by maxamillion** [comment]
@mkrizek I think this should be backported to `stable-2.7` once merged, do you agree?

**Comment by mkrizek** [comment]
@maxamillion Created https://github.com/ansible/ansible/pull/47414.

**Pull request merged**